src/interfaces/design.d.ts

```diff
@@ -43,6 +43,7 @@ export interface ICourtColor {
 }
 
 export interface ISaveDesign {
+  designer?: string;

```

Comment:
what is case that _designer_ is undefined?

Reply:
> what is case that _designer_ is undefined?

This interface will be called in both creating template and generating shopping cart item. And the designer field will be only used to create the new template.